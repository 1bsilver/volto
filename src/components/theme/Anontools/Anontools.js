/**
 * Anontools component.
 * @module components/theme/Anontools/Anontools
 */

import React, { PropTypes, Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';

@connect(
  state => ({
    token: state.userSession.token,
  }),
)
/**
 * Anontools container class.
 * @class Anontools
 * @extends Component
 */
export default class Anontools extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    token: PropTypes.string,
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      !this.props.token &&
        <div id="portal-anontools">
          <ul>
            <li>
              <Link to="/login" id="personaltools-login">Log in</Link>
            </li>
          </ul>
        </div>
    );
  }
}
