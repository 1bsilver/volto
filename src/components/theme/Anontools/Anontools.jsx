/**
 * Anontools component.
 * @module components/theme/Anontools/Anontools
 */

import React, { PropTypes, Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { List } from 'semantic-ui-react';

@connect(
  state => ({
    token: state.userSession.token,
  }),
)
/**
 * Anontools container class.
 * @class Anontools
 * @extends Component
 */
export default class Anontools extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    token: PropTypes.string,
  }

  /**
   * Default properties.
   * @property {Object} defaultProps Default properties.
   * @static
   */
  static defaultProps = {
    token: null,
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      !this.props.token &&
        <List floated="right" horizontal>
          <Link className="item" to="/login">Log in</Link>
        </List>
    );
  }
}
