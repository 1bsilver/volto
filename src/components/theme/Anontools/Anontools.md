```jsx noeditor
import Wrapper from '@plone/volto/styleguide';

<Wrapper anonymous={true}>
  <Anontools token="" content="" />
</Wrapper>
```
