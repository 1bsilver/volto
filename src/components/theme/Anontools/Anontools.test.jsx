import React from 'react';
import renderer from 'react-test-renderer';
import Anontools from './Anontools';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore();

describe('Anontools', () => {
  it('renders an anontools component when no token is specified', () => {
    const store = mockStore({ userSession: { token: null } });
    const component = renderer.create(
      <Provider store={store}>
        <Anontools />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should not render an anontools component when a token is specified', () => {
    const store = mockStore({ userSession: { token: '1234' } });
    const component = renderer.create(
      <Provider store={store}>
        <Anontools />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
