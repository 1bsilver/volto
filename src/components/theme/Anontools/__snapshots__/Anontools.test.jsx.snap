// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anontools renders an anontools component when no token is specified 1`] = `
<div
  id="portal-anontools"
>
  <ul>
    <li>
      <a
        id="personaltools-login"
        onClick={[Function]}
        style={Object {}}
      >
        Log in
      </a>
    </li>
  </ul>
</div>
`;

exports[`Anontools should not render an anontools component when a token is specified 1`] = `null`;
